### PR TITLE
Initial deployment of End-to-end testing to run on schedule

### PR DESCRIPTION
**Pull request by angeloskyratzakos**
#### Summary
Changes to allow the creation and the destruction of cluster and the deployment of provisioner from scratch 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24524

### CODE REVIEW

File: terraform/aws/modules/cluster/aws_auth_configmap.tf [hunk 1]
```diff
@@ -17,15 +17,20 @@ resource "kubernetes_config_map" "aws_auth_configmap" {
...
+}
+
+resource "time_sleep" "wait_20_seconds" {
   depends_on = [
     aws_eks_cluster.cluster,
     null_resource.cluster_services,
     aws_autoscaling_group.worker-asg,
     aws_iam_role.lambda_role
   ]
+  create_duration = "20s"
```
Comment (angeloskyratzakos) [inline terraform/aws/modules/cluster/aws_auth_configmap.tf:33]: Adding a sleep resource as sometimes randomly the auth configmap was failing

File: terraform/aws/modules/cluster/kubeconfig.tf [hunk 2]
```diff
@@ -8,16 +8,17 @@ data "template_file" "kubeconfig" {
...
-      echo $KUBECONFIG_DATA | base64 --decode > "${var.kubeconfig_dir}/kubeconfig"
+      echo $KUBECONFIG_DATA > "${var.kubeconfig_dir}/kubeconfig"
+      cat "${var.kubeconfig_dir}/kubeconfig"
     LOCAL_EXEC
     environment = {
-      KUBECONFIG_DATA = "${base64encode(data.template_file.kubeconfig.rendered)}"
+      KUBECONFIG_DATA = "${data.template_file.kubeconfig.rendered}"
     }
   }
 }
```
Comment (angeloskyratzakos) [inline terraform/aws/modules/cluster/kubeconfig.tf:24]: do we need that resource?

File: terraform/aws/modules/provisioner/providers.tf [hunk 3]
```diff
@@ -0,0 +1,21 @@
+provider "kubernetes" {
+  host                   = data.aws_eks_cluster.cluster.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
+  token                  = data.aws_eks_cluster_auth.cluster_auth.token
+  load_config_file       = false
+  # config_path            = "${var.kubeconfig_dir}/kubeconfig"
```
Comment (angeloskyratzakos) [inline terraform/aws/modules/provisioner/providers.tf:6]: do we need the config path? as we load the config on the fly

File: terraform/aws/modules/provisioner/provisioner.tf [hunk 4]
```diff
@@ -77,7 +77,7 @@ resource "kubernetes_deployment" "mattermost_cloud_main" {
         container {
           name  = "mattermost-cloud"
           image = var.mattermost_cloud_image
-          args  = ["server", "--debug", "--state-store", "mattermost-kops-state-${var.environment}", "--private-dns", "$(PRIVATE_DNS)", "--keep-filestore-data=$(KEEP_FILESTORE_DATA)", "--keep-database-data=$(KEEP_DATABASE_DATA)", "--database", "$(DATABASE)"]
+          args  = ["server", "--debug", "--machine-readable-logs", "--cluster-installation-supervisor=false", "--installation-supervisor=false", "--state-store", "mattermost-kops-state-${var.environment}", "--keep-filestore-data=$(KEEP_FILESTORE_DATA)", "--keep-database-data=$(KEEP_DATABASE_DATA)", "--database", "$(DATABASE)"]
```
Comment (angeloskyratzakos) [inline terraform/aws/modules/provisioner/provisioner.tf:80]: updates to bring it up to date

File: terraform/aws/modules/provisioner/providers.tf (hunk 3 above)
Comment (stylianosrigas) [reply terraform/aws/modules/provisioner/providers.tf:6]: if we don't use the config path it means that when we run locally we always need to authenticate first right?

File: terraform/aws/modules/provisioner/providers.tf [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+provider "helm" {
```
Comment (stylianosrigas) [inline terraform/aws/modules/provisioner/providers.tf:9]: why we need a new Helm provider here?

File: terraform/aws/modules/cluster/aws_auth_configmap.tf (hunk 1 above)
Comment (angeloskyratzakos) [reply terraform/aws/modules/cluster/aws_auth_configmap.tf:33]: will increase it to 40s as it still failed once

**Comment by cpanato** [comment]
ps. dont forget to squash the commits

**Comment by angeloskyratzakos** [comment]
@cpanato it works, but it needs some fine tuning, to run several times to make sure that it works from scratch

File: terraform/aws/modules/provisioner/providers.tf (hunk 5 above)
Comment (angeloskyratzakos) [reply terraform/aws/modules/provisioner/providers.tf]: removed it

**Comment by cpanato** [comment]
@angeloskyratzakos can you squash the commits?

**Review by cpanato — APPROVED** [review]
lgtm this initial changes, thanks for working on that

can you squash the commits?

**Comment by angeloskyratzakos** [comment]
> @angeloskyratzakos can you squash the commits?



> lgtm this initial changes, thanks for working on that
> 
> can you squash the commits?

I will squash them before merging in order to avoid force push, or should I force push?

**Comment by cpanato** [comment]
you can force push, that is not a problem on that

**Review by stylianosrigas — APPROVED** [review]
LGTM

**Review by cpanato — APPROVED** [review]

**Pull request merged**